### PR TITLE
enable specifying weights path for fid

### PR DESCRIPTION
**Pull request by yuvalkirstain**
## What does this PR do?

Fixes #\<issue_number>

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

In some places compute nodes do have access to internet, so it is important to enable loading metrics without internet access. This PR enables specifying the checkpoint path for FID.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2867.org.readthedocs.build/en/2867/

### CODE REVIEW

File: src/torchmetrics/image/fid.py [hunk 1]
```diff
@@ -322,7 +323,11 @@ def __init__(
                     f"Integer input to argument `feature` must be one of {valid_int_input}, but got {feature}."
                 )
 
-            self.inception = NoTrainInceptionV3(name="inception-v3-compat", features_list=[str(feature)])
+            self.inception = NoTrainInceptionV3(
```
Comment (Borda) [inline src/torchmetrics/image/fid.py:326]: you need to provide a default to preserve actual behavior...

**Review by Borda — COMMENTED** [review]
I git it but still pls share motivation for this PR

**Comment by yuvalkirstain** [comment]
I did (perhaps not in the right place) - "In some places compute nodes do have access to internet, so it is important to enable loading metrics without internet access. This PR enables specifying the checkpoint path for FID."

File: src/torchmetrics/image/fid.py (hunk 1 above)
Comment (yuvalkirstain) [reply src/torchmetrics/image/fid.py:326]: The default is None and it preserves the actual behavior:
See https://github.com/toshas/torch-fidelity/blob/master/torch_fidelity/feature_extractor_inceptionv3.py#L31

**Review by Borda — APPROVED** [review]

**Review by lantiga — APPROVED** [review]
Thanks!

**Pull request merged**